opte/Cargo.toml

```diff
@@ -11,7 +11,7 @@ engine = ["api"]
 # This feature indicates that OPTE is being built with std. This is
 # mostly useful to consumers of the API, providing convenient methods
 # for working with the API types in a std context.
-std = []
+std = ["opte-api/std"]
```

Comment:
Is there any consumer of the `opte-api` crate that does _not_ use the `std` feature? There are some types that are duplicated with those in `std`, like the IP addresses. I think that came initially from defining those types directly in the `opte` core crate or some other no-std environment, but we might reduce maintenance by using those `std` types directly if we can get away with it.

Reply:
Yes, the API must be shared with clients as well as the engine, as these types are how clients send requests to the engine.

Reply:
Oh, I just misread this as a default feature. I understand the API types are shared between consumers and the engine, I think I just convinced myself that the engine is built _with_ this feature, which it may not be.

Reply:
You are actually onto something here. Now that the API types are their own crate the only real reason to build opte with `std` is for testing. However, there was also the historical idea that perhaps opte would also run in contexts other than as part of a mac provider like `xde`. Going back to the conversations we had around concurrency control, we might want to put a stake in the ground and say that opte is only ever meant to run in that type of context. Or perhaps we want to keep it flexible and just state that whatever is driving the opte engine needs to enforce concurrency control. But this is all opening a can of worms I'm trying to keep closed for the moment, or at least well outside the scope of this change. That said, now that the API types are in their own crate it might make sense to at least not have `std + api` as defaults, but I'll leave that to its own issue/PR if we decide that feels better.

Reply:
That all makes sense, and putting this off to another PR after more discussion sounds solid. It also answers the original question, with "Yes, we still need types that work in both std and no-std environments".